lldb/include/lldb/Core/SwiftScratchContextReader.h

```diff
@@ -97,27 +47,25 @@ struct ScopedSharedMutexReader {
 /// or even separate scratch contexts for each lldb::Module. But it
 /// will only do this if no client holds on to a read lock on \b
 /// m_scratch_typesystem_lock.
-class SwiftScratchContextReader : ScopedSharedMutexReader {
-  TypeSystemSwiftTypeRefForExpressions *m_ptr;
+class SwiftScratchContextReader {
+  std::shared_lock<std::shared_mutex> m_lock;
```

Comment:
All the locks of the shared mutex are locked by `std::shared_lock` (none of them use `std::unique_lock`. Doesn't this mean that all of them can use the type system concurrently freely?

Reply:
Correct. The lock is not meant to ensure exclusive access, but guard against a context from being destroyed while still being used.

Reply:
Fair enough, you would have to release the unique_lock and acquire the shared_lock, which has the same drawback as the current approach.